draft-ietf-dnsop-svcb-https.md

```diff
@@ -659,6 +661,27 @@ each RRSet in the Additional section with the same DNSSEC-related records
 that they would send when providing that RRSet as an Answer (e.g. RRSIG, NSEC,
 NSEC3).
 
+## EDNS Client Subnet (ECS) {#ecs}
+
+The EDNS Client Subnet option (ECS, {{!RFC7871}}) allows recursive
+resolvers to request IP addresses that are suitable for a particular client
+IP range.  SVCB records may contain IP addresses (in ipv*hint SvcParams),
+or direct users to a subnet-specific TargetName, so recursive resolvers
+SHOULD include the same ECS option in SVCB queries as in A/AAAA queries.
+
+According to Section 7.3.1 of {{!RFC7871}}, "Any records from \[the
+Additional section\] MUST NOT be tied to a network".  Accordingly,
+resolvers SHOULD treat any records in the Additional section as having
+SOURCE PREFIX-LENGTH zero and SCOPE PREFIX-LENGTH as specified
+in the ECS option, and MAY cache them on this basis.  Authoritative
```

Comment:
https://tools.ietf.org/html/rfc2181#section-5.4.1 recommends not forwarding Additional section records to clients.

_**Unauthenticated RRs received and cached from the least trustworthy of
those groupings, that is data from the additional data section, and
data from the authority section of a non-authoritative answer, should
not be cached in such a way that they would ever be returned as
answers to a received query.  They may be returned as additional
information where appropriate.  Ignoring this would allow the
trustworthiness of relatively untrustworthy data to be increased
without cause or excuse.**_

Therefore the recommendation:
_**Accordingly,
resolvers SHOULD treat any records in the Additional section as having
SOURCE PREFIX-LENGTH zero and SCOPE PREFIX-LENGTH as specified
in the ECS option, and MAY cache them on this basis.**_

is unlikely to have an impact in the real world - popular DNS resolvers/implementations (e.g. Quad{189}, OpenDNS) do not propagate the ECS information to their clients (stubs mentioned here). This can be seen in the handling of MX records.

$ dig @ns1.google.com -t MX google.com +nostats
<snip>
;; ANSWER SECTION:
google.com.		600	IN	MX	20 alt1.aspmx.l.google.com.
google.com.		600	IN	MX	30 alt2.aspmx.l.google.com.
google.com.		600	IN	MX	50 alt4.aspmx.l.google.com.
google.com.		600	IN	MX	10 aspmx.l.google.com.
google.com.		600	IN	MX	40 alt3.aspmx.l.google.com.

;; ADDITIONAL SECTION:
alt1.aspmx.l.google.com. 293	IN	A	173.194.216.26
alt1.aspmx.l.google.com. 293	IN	AAAA	2607:f8b0:400c:c12::1a
alt2.aspmx.l.google.com. 293	IN	A	64.233.186.26
alt2.aspmx.l.google.com. 293	IN	AAAA	2800:3f0:4003:c00::1a
alt4.aspmx.l.google.com. 293	IN	A	172.253.120.26
alt4.aspmx.l.google.com. 293	IN	AAAA	2a00:1450:400c:c01::1a
aspmx.l.google.com.	293	IN	A	142.250.123.26
aspmx.l.google.com.	293	IN	AAAA	2607:f8b0:4023:1402::1a
alt3.aspmx.l.google.com. 293	IN	A	209.85.203.26
alt3.aspmx.l.google.com. 293	IN	AAAA	2a00:1450:400b:c03::1a

$ dig @9.9.9.9 -t MX google.com +nostats
<snip>
;; ANSWER SECTION:
google.com.		600	IN	MX	10 aspmx.l.google.com.
google.com.		600	IN	MX	20 alt1.aspmx.l.google.com.
google.com.		600	IN	MX	40 alt3.aspmx.l.google.com.
google.com.		600	IN	MX	50 alt4.aspmx.l.google.com.
google.com.		600	IN	MX	30 alt2.aspmx.l.google.com.

Reply:
Thanks for investigating that.  I didn't know about RFC 2181.

I think the text you quoted is (surprisingly?) in line with this draft.

> **Unauthenticated** RRs ... from the additional data section ... should
> not be cached in such a way that they would ever be returned as
> answers

SVCB Section 4.1:
> If the zone is signed, the server SHOULD also include positive or negative DNSSEC responses for these records in the Additional section.

So these Additional records could actually end up cached and used for general purposes.

RFC 2181
> They may be returned as additional information where appropriate.

SVCB Section 4.2:
> ... adding all useful RRs from the response to the Additional section

So even without authentication, we're using these records as recommended in RFC 2181.

Thanks for showing an easy way to check the real world behavior.
> ... popular DNS resolvers/implementations (e.g. Quad{189}, OpenDNS) do not propagate the ECS information to their clients

I think they do not propagate any Additional data to their clients.  That seems like a compliant, but perhaps not highly optimized behavior.  A highly optimized resolver could comply with RFC 2181 by maintaining a separate cache for records returned in the Additional section, to be used only when populating the Additional section of responses (when appropriate).  If that cache assumes SCOPE /0, only global records can go in the Additional section, whereas if it assumes SOURCE /0, authoritative servers can safely return ECS-tailored Additional records.

As a practical matter this may be moot in current implementations, but as a matter of requirements I think the text here still makes sense.

Reply:
To provide context to a reader, will be be good to have a reference to https://tools.ietf.org/html/rfc2181#section-5.4.1 after the phrase "and MAY cache them on this basis".

Reply:
OK.  The caching question is mostly independent of ECS, so I moved it to the preceding section and added the full quote from RFC 2181.